pkg/planner/cardinality/row_count_index.go

```diff
@@ -61,16 +63,26 @@ func GetRowCountByIndexRanges(sctx planctx.PlanContext, coll *statistics.HistCol
 		}
 	}
 	recordUsedItemStatsStatus(sctx, idx, coll.PhysicalID, idxID)
+	corrResult = float64(0)
 	if statistics.IndexStatsIsInvalid(sctx, idx, coll, idxID) {
-		colsLen := -1
-		if idx != nil && idx.Info.Unique {
-			colsLen = len(idx.Info.Columns)
-		}
-		result, err = getPseudoRowCountByIndexRanges(sc.TypeCtx(), indexRanges, float64(coll.RealtimeCount), colsLen)
-		if err == nil && sc.EnableOptimizerCETrace && idx != nil {
-			ceTraceRange(sctx, coll.PhysicalID, colNames, indexRanges, "Index Stats-Pseudo", uint64(result))
+		if hasColumnStats(sctx, coll, idxCols) {
+			// If it is a single column index and there are column stats, use column estimation instead.
+			if len(idxCols) == 1 {
```

Comment:
feels like the len(idxCols) == 1 case can also be handled in getPseudoRowCountWithPartialStats, do we need to special handle it here?